src/imitation/data/types.py

```diff
@@ -298,24 +297,105 @@ def __post_init__(self):
         _rews_validation(self.rews, self.acts)
 
 
-def load(path: AnyPath) -> Sequence[TrajectoryWithRew]:
+def load_with_rewards(path: AnyPath) -> Sequence[TrajectoryWithRew]:
+    """Loads a sequence of trajectories with rewards from a file."""
+    data = load(path)
+    if not all(isinstance(traj, TrajectoryWithRew) for traj in data):
+        raise ValueError(
+            f"Expected all trajectories to be of type `TrajectoryWithRew`, "
+            f"but found {type(data).__name__}",
+        )
+
+    return cast(Sequence[TrajectoryWithRew], data)
+
+
+def load(path: AnyPath) -> Sequence[Trajectory]:
     """Loads a sequence of trajectories saved by `save()` from `path`."""
-    with open(path, "rb") as f:
-        return pickle.load(f)
+    # Interestingly, np.load will just silently load a normal pickle file when you
+    # set `allow_pickle=True`. So this call should succeed for both the new compressed
+    # .npz format and the old pickle based format. To tell the difference we need to
+    # look at the type of the resulting object. If it's the new compressed format,
+    # it should be a Mapping that we need to decode, whereas if it's the old format
+    # it's just the sequence of trajectories and we can return it directly.
+    data = np.load(path, allow_pickle=True)
+    if isinstance(data, Sequence):
+        return data
+
```

Comment:
I guess it just looked a bit nicer to my eye as I was writing it to have an extra blank line there, but I'll go ahead and remove it. As a side note, do you have any preference for whether to use `elif` vs. just a bare `if` after an `if`-block that ends in a `return` statement like this? I think my usual habit is to use `elif` since it's a bit more explicit and consistent but I'm happy to follow whatever convention you prefer.